src/tutorial/testing.md

```diff
@@ -1,5 +1,101 @@
 # Testing
 
+Over the decades of people doing software development
+one truth has been found:
+Untested software rarely works.
+(Many people would go so far and add
+"Most tested software doesn't work either."
+But we are all optimists here, right?)
+So, to ensure that your program does what you expect it to do,
+it is wise to test it.
+
+One easy way to do that is
+to write a `README` file
+that describes what your program should do.
+And when you feel ready to make a new release,
+go through the `README` and ensure that
+the behavior is still as expected.
+You can make this a more rigorous exercise
+by also writing down how your program should react to erroneous inputs.
+
+Here's another fancy idea:
+Write that `README` before you write the code.
+
+<aside>
+
+**Aside:**
+Have a look at
+[Test-driven development](https://en.wikipedia.org/wiki/Test-driven_development) (TDD)
+if you haven't heard of it.
+
+</aside>
+
+## Automated testing
+
+Now, this is all fine and dandy,
+but doing all of this manually?
+That can take a lot of time.
+I don't know about you,
```

Comment:
nit: maybe there is a way to rephrase it to not speak from first person tense in the book?

Particularly here it looks like you can just keep:

>Let's talk about how to automate these tests.

This feeling when you write something (book, docs, code) and want to left some personal touch, yea :)

Reply:
Thanks -- I really wasn't sure about this, either. I know I can just omit the whole sentence… but how about this?

> Now, this is all fine and dandy,
> but doing all of this manually?
> That can take a lot of time.
> _At the same time,
> many people have come to enjoy telling computers to do things for them._
> Let's talk about how to automate these tests.

Reply:
lgtm